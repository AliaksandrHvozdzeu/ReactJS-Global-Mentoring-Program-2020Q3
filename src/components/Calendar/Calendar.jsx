import React from 'react';
import PropTypes from 'prop-types';

export default function Calendar({ id, title, value, onChange }) {

  return (
    <>
      <label htmlFor={id}>
        {title}
        <input className="input" id={id} type="date" value={value} onChange={onChange}/>
      </label>
    </>
  );

}

Calendar.propTypes = {
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  title: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};
