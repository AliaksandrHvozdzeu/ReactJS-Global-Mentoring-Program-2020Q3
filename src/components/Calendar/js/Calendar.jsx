import React from 'react';
import PropTypes from 'prop-types';
import ErrorMessage from '../../ErrorMessage';

export default function Calendar({ id, title, value, onChange, name, type, onBlur, error }) {

  return (
    <>
      <label htmlFor={id}>
        {title}
        <input id={id}
               className="input"
               type={type}
               name={name}
               value={value}
               onChange={onChange}
               onBlur={onBlur}/>
      </label>
      <ErrorMessage error={error}/>
    </>
  );

}

Calendar.propTypes = {
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  title: PropTypes.string.isRequired,
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  name: PropTypes.string,
  type: PropTypes.string.isRequired,
  error: PropTypes.string.isRequired,
  onBlur: PropTypes.func.isRequired,
};

Calendar.defaultProps = {
  value: '',
  name: '',
}
