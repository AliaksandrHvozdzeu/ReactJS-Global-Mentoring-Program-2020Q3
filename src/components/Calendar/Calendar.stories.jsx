import React from "react";
import { withKnobs } from "@storybook/addon-knobs";
import Calendar from "./js/Calendar";
import "./css/Calendar.css";
import "../../App.css";

export default {
  title: "Calendar",
  decorators: [withKnobs]
};

export const Default = () => (
  <>
    <label htmlFor="calendar_id">
      Calendar title
      <input id="calendar_id"
             className="input"
             type="calendar"
             name="calendar"
             value="10.10.2020"
             onChange={() => {
               alert("onChange");
             }}
             onBlur />
    </label>
  </>
);
