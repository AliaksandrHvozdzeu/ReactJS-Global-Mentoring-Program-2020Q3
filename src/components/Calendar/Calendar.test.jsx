import React from 'react';
import rerender from 'react-test-renderer';
import Calendar from './js/Calendar';

describe('when Calendar', () => {
  test('then snapshot created', () => {
    const id = 'id';
    const title = 'title';
    const value = 'value';
    const onChange = () => {
    };
    const name = 'name';
    const type = 'type';
    const onBlur = () => {
    };
    const error = 'error';
    const component = rerender.create(<Calendar id={id}
                                                title={title}
                                                value={value}
                                                onChange={onChange}
                                                name={name}
                                                type={type}
                                                onBlur={onBlur}
                                                error={error}/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
