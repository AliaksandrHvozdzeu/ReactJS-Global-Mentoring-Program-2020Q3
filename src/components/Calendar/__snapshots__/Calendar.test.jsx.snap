// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when Calendar then snapshot created 1`] = `
Array [
  <label
    htmlFor="id"
  >
    title
    <input
      className="input"
      id="id"
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
      type="type"
      value="value"
    />
  </label>,
  <p
    className="input-error"
  >
    error
  </p>,
]
`;
